[DOCS] Rework conceptual info for ILM. (#52181)
elastic/elasticsearch: docs/reference/ilm/actions/ilm-forcemerge.asciidoc

[[ilm-forcemerge]]
=== Force Merge

Phases allowed: hot, warm.

> NOTE: Index will be be made read-only when this action is run
> (see: <<dynamic-index-settings,index.blocks.write>>)

Reviewer: ```suggestion
NOTE: The index is made read-only when this action is run
(see: <<dynamic-index-settings,index.blocks.write>>)
```

NOTE: If the `forcemerge` action is used in the `hot` phase, the `rollover` action *must* be preset.

> ILM validates this predicate and will refuse a policy with a forcemerge in the hot phase without a

Reviewer: ```suggestion
{ilm-init} validates this predicate and will refuse a policy with a forcemerge in the hot phase without a
```

> rollover action.

Reviewer: Not a huge fan of stacking admonitions, but it may be necessary here. Should probably move them after the action description though.

Author: Agree we should avoid stack admonitions whenever possible.

> The Force Merge Action <<indices-forcemerge,force merges>> the index into at
> most a specific number of <<indices-segments,segments>>.

Reviewer: ```suggestion
The force merge action <<indices-forcemerge,force merges>> the index into a
specified maximum number of <<indices-segments,segments>>.
```

[[ilm-forcemerge-options]]
.Force Merge Options
[options="header"]
|======
| Name                 | Required  | Default             | Description
| `max_num_segments`   | yes       | -                   | The number of segments to merge to. To fully merge the index, set it to `1`
| `codec`              | no        | -                   | Optional specification of the `best_compression` codec

> |======

Reviewer: See previous comment re: heading and converting to a def list.

[WARNING]
======

> When using the `"codec": "best_compression"` configuration in the ILM forcemerge action, ILM will

Reviewer: ```suggestion
When using the `"codec": "best_compression"` configuration in the {ilm-init} forcemerge action, {ilm-int} will
```

<<indices-close,close>> and then <<indices-open-close,re-open>> the index prior to a forcemerge.
During this time the index will be unavailable for either read or write operations.
======

> 

Reviewer: I'd add an `Examples` heading/section.
